packages/service-library/src/servicelib/logging_utils.py

```diff
@@ -179,43 +192,156 @@ def config_all_loggers(
+        log_format_local_dev_enabled=log_format_local_dev_enabled,
+    )
+
+    # Create a properly formatted handler for the root logger
+    root_handler = logging.StreamHandler()
+    root_handler.setFormatter(
+        CustomFormatter(fmt, log_format_local_dev_enabled=log_format_local_dev_enabled)
+    )
+
+    all_loggers = _get_all_loggers()
+
+    # Apply comprehensive logging setup
+    # Note: We don't store the original state here since this is a permanent setup
+    _apply_comprehensive_logging_setup(all_loggers, root_handler)
 
+    # Apply filters
+    _apply_logger_filters(logger_filter_mapping)
 
-def test_logger_propagation(logger: logging.Logger) -> None:
-    """log propagation and levels can sometimes be daunting to get it right.
 
-    This function uses the `logger`` passed as argument to log the same message at different levels
+@contextmanager
+def async_loggers(
+    *,
+    log_format_local_dev_enabled: bool,
+    logger_filter_mapping: dict[LoggerName, list[MessageSubstring]],
+    tracing_settings: TracingSettings | None,
+    log_base_level: LogLevelInt,
+    noisy_loggers: tuple[str, ...] | None,
+) -> Iterator[None]:
+    """
+    context manager for non-blocking logging infrastructure.
 
-    This should help to visually test a given configuration
+    Usage:
+        with async_loggers(log_format_local_dev_enabled=True, logger_filter_mapping={}, tracing_settings=None):
+            # Your async application code here
+            logger.info("This is non-blocking!")
 
-    USAGE:
-        from .logging_utils import test_logger_propagation
-        for n in ("aiohttp.access", "gunicorn.access"):
-            test_logger_propagation(logging.getLogger(n))
+    Args:
+        log_format_local_dev_enabled: Enable local development formatting
+        logger_filter_mapping: Mapping of logger names to filtered message substrings
+        tracing_settings: OpenTelemetry tracing configuration
     """
-    msg = f"TESTING %s log using {logger=}"
-    logger.critical(msg, "critical")
-    logger.error(msg, "error")
-    logger.info(msg, "info")
-    logger.warning(msg, "warning")
-    logger.debug(msg, "debug")
+    _setup_base_logging_level(log_base_level)
+    if noisy_loggers is not None:
+        _dampen_noisy_loggers(noisy_loggers)
+
+    if tracing_settings is not None:
+        setup_log_tracing(tracing_settings=tracing_settings)
+    fmt = _setup_format_string(
+        tracing_settings=tracing_settings,
+        log_format_local_dev_enabled=log_format_local_dev_enabled,
+    )
+
+    # Set up async logging infrastructure
+    log_queue: queue.Queue[logging.LogRecord] = queue.Queue()
+    # Create handler with proper formatting
+    handler = logging.StreamHandler()
+    handler.setFormatter(
+        CustomFormatter(fmt, log_format_local_dev_enabled=log_format_local_dev_enabled)
+    )
+
+    # Create and start the queue listener
+    listener = logging.handlers.QueueListener(
```

Comment:
this listener is  trhead, right?

Reply:
please see:
- https://docs.python.org/3/library/queue.html#queue-objects
- https://docs.python.org/3/library/logging.handlers.html#queuehandler